src/shell.js

```diff
@@ -194,7 +194,7 @@ var requireNodeFS;
 if (ENVIRONMENT_IS_NODE) {
 #if ENVIRONMENT
 #if ASSERTIONS
-  if (!(typeof process == 'object' && typeof require == 'function')) throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
+  if (process.release.name !== 'node') throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
```

Comment:
I think you need check for `process` and `process.release` before you can check for `process.release.name` don't you?

Reply:
Can we use `?.` in Emscripten yet?

Reply:
Sadly it doesn't look like it.. at least not by default: https://caniuse.com/mdn-javascript_operators_optional_chaining

For example our min chrome version is 75 by default and this syntax came along on 80.

Reply:
How about `if (!global.process || !global.process.release || global.process.release.name !== 'node' ...`

Reply:
@sbc100 Good catch thank you! Fixed with [aea257b](https://github.com/emscripten-core/emscripten/pull/17661/commits/aea257b57fbaf8becea577399e3c236cb1ab1970).